<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1512</width>
    <height>806</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <widget class="QWidget" name="data_widget" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QTabWidget" name="tabWidget">
         <property name="currentIndex">
          <number>2</number>
         </property>
         <widget class="QWidget" name="raw_data">
          <attribute name="title">
           <string>Raw Data</string>
          </attribute>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QWidget" name="widget_4" native="true">
             <property name="maximumSize">
              <size>
               <width>250</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QWidget" name="widget_3" native="true">
                <widget class="QLabel" name="label">
                 <property name="geometry">
                  <rect>
                   <x>0</x>
                   <y>10</y>
                   <width>58</width>
                   <height>16</height>
                  </rect>
                 </property>
                 <property name="text">
                  <string>Filename:</string>
                 </property>
                 <property name="buddy">
                  <cstring>filename_name</cstring>
                 </property>
                </widget>
                <widget class="QLineEdit" name="filename_name">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="geometry">
                  <rect>
                   <x>70</x>
                   <y>10</y>
                   <width>150</width>
                   <height>20</height>
                  </rect>
                 </property>
                 <property name="minimumSize">
                  <size>
                   <width>150</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>50</width>
                   <height>20</height>
                  </size>
                 </property>
                </widget>
                <widget class="QPushButton" name="add_roi_button">
                 <property name="geometry">
                  <rect>
                   <x>-10</x>
                   <y>40</y>
                   <width>238</width>
                   <height>32</height>
                  </rect>
                 </property>
                 <property name="toolTip">
                  <string>Adds a ROI box to the image to perform a quick visualization</string>
                 </property>
                 <property name="text">
                  <string>Add Roi</string>
                 </property>
                </widget>
                <widget class="QPushButton" name="plot_roi_button">
                 <property name="geometry">
                  <rect>
                   <x>-10</x>
                   <y>80</y>
                   <width>238</width>
                   <height>32</height>
                  </rect>
                 </property>
                 <property name="toolTip">
                  <string>Plot the signal, FFT, and spectrogram on the selected ROI</string>
                 </property>
                 <property name="text">
                  <string>Plot</string>
                 </property>
                </widget>
                <widget class="QGroupBox" name="groupBox_5">
                 <property name="geometry">
                  <rect>
                   <x>0</x>
                   <y>130</y>
                   <width>226</width>
                   <height>291</height>
                  </rect>
                 </property>
                 <property name="title">
                  <string>Camera</string>
                 </property>
                 <layout class="QFormLayout" name="formLayout_5">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_32">
                    <property name="text">
                     <string>Exposure</string>
                    </property>
                    <property name="buddy">
                     <cstring>line_exposure</cstring>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QLineEdit" name="exposure_freerun"/>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="label_33">
                    <property name="text">
                     <string>Gain</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QLineEdit" name="gain_freerun"/>
                  </item>
                  <item row="2" column="0">
                   <widget class="QLabel" name="label_34">
                    <property name="text">
                     <string>Frame Rate</string>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="1">
                   <widget class="QLineEdit" name="framerate_freerun"/>
                  </item>
                  <item row="3" column="0">
                   <widget class="QLabel" name="label_35">
                    <property name="text">
                     <string>Width</string>
                    </property>
                   </widget>
                  </item>
                  <item row="3" column="1">
                   <widget class="QLineEdit" name="width_freerun"/>
                  </item>
                  <item row="4" column="0">
                   <widget class="QLabel" name="label_36">
                    <property name="text">
                     <string>Height</string>
                    </property>
                   </widget>
                  </item>
                  <item row="4" column="1">
                   <widget class="QLineEdit" name="height_freerun"/>
                  </item>
                  <item row="5" column="0">
                   <widget class="QLabel" name="label_37">
                    <property name="text">
                     <string>X-Center</string>
                    </property>
                   </widget>
                  </item>
                  <item row="5" column="1">
                   <widget class="QLineEdit" name="xcenter_freerun"/>
                  </item>
                  <item row="6" column="0">
                   <widget class="QLabel" name="label_38">
                    <property name="text">
                     <string>Y-Center</string>
                    </property>
                   </widget>
                  </item>
                  <item row="6" column="1">
                   <widget class="QLineEdit" name="ycenter_freerun"/>
                  </item>
                  <item row="7" column="0">
                   <widget class="QLabel" name="label_19">
                    <property name="text">
                     <string>Autogain</string>
                    </property>
                   </widget>
                  </item>
                  <item row="8" column="0">
                   <widget class="QLabel" name="label_20">
                    <property name="text">
                     <string>Pixel Format:</string>
                    </property>
                   </widget>
                  </item>
                  <item row="8" column="1">
                   <widget class="QComboBox" name="pixel_format_box"/>
                  </item>
                 </layout>
                </widget>
                <widget class="QCheckBox" name="freerun_checkbox">
                 <property name="geometry">
                  <rect>
                   <x>80</x>
                   <y>500</y>
                   <width>72</width>
                   <height>20</height>
                  </rect>
                 </property>
                 <property name="text">
                  <string>Freerun</string>
                 </property>
                </widget>
                <widget class="QLineEdit" name="refresh_rate_value">
                 <property name="geometry">
                  <rect>
                   <x>60</x>
                   <y>470</y>
                   <width>119</width>
                   <height>21</height>
                  </rect>
                 </property>
                </widget>
                <widget class="QLabel" name="label_18">
                 <property name="geometry">
                  <rect>
                   <x>20</x>
                   <y>470</y>
                   <width>60</width>
                   <height>16</height>
                  </rect>
                 </property>
                 <property name="text">
                  <string>FPS</string>
                 </property>
                </widget>
                <widget class="QPushButton" name="update_config">
                 <property name="geometry">
                  <rect>
                   <x>20</x>
                   <y>430</y>
                   <width>181</width>
                   <height>32</height>
                  </rect>
                 </property>
                 <property name="text">
                  <string>Update Configuration</string>
                 </property>
                </widget>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="fft_widget">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <attribute name="title">
           <string>Fourier Data</string>
          </attribute>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QWidget" name="widget_5" native="true">
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>16777215</height>
              </size>
             </property>
             <widget class="QLabel" name="label_2">
              <property name="geometry">
               <rect>
                <x>13</x>
                <y>12</y>
                <width>92</width>
                <height>16</height>
               </rect>
              </property>
              <property name="text">
               <string>Filter Freq (Hz)</string>
              </property>
              <property name="buddy">
               <cstring>frequency_line</cstring>
              </property>
             </widget>
             <widget class="QLineEdit" name="frequency_line">
              <property name="geometry">
               <rect>
                <x>113</x>
                <y>12</y>
                <width>75</width>
                <height>21</height>
               </rect>
              </property>
              <property name="toolTip">
               <string>Frequency to filter the data after a Fourier Transform</string>
              </property>
             </widget>
             <widget class="QLabel" name="label_3">
              <property name="geometry">
               <rect>
                <x>40</x>
                <y>43</y>
                <width>65</width>
                <height>16</height>
               </rect>
              </property>
              <property name="text">
               <string>Min cycles</string>
              </property>
              <property name="buddy">
               <cstring>min_cycles_line</cstring>
              </property>
             </widget>
             <widget class="QLineEdit" name="min_cycles_line">
              <property name="geometry">
               <rect>
                <x>113</x>
                <y>43</y>
                <width>75</width>
                <height>21</height>
               </rect>
              </property>
              <property name="toolTip">
               <string>For how many cycles should the signal be integrated before calculating the FFT</string>
              </property>
             </widget>
             <widget class="QPushButton" name="filter_data_button">
              <property name="geometry">
               <rect>
                <x>6</x>
                <y>74</y>
                <width>105</width>
                <height>32</height>
               </rect>
              </property>
              <property name="toolTip">
               <string>Use the frequency and number of cycles to perform an FFT over all the images</string>
              </property>
              <property name="text">
               <string>Filter Data</string>
              </property>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="live_data_widget">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <attribute name="title">
           <string>Live Data</string>
          </attribute>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QWidget" name="control_widget" native="true">
             <property name="maximumSize">
              <size>
               <width>250</width>
               <height>16777215</height>
              </size>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <widget class="QPushButton" name="button_config_edit">
                <property name="text">
                 <string>Config Edit</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBox">
                <property name="title">
                 <string>Camera</string>
                </property>
                <layout class="QFormLayout" name="formLayout">
                 <item row="0" column="0">
                  <widget class="QLabel" name="label_4">
                   <property name="text">
                    <string>Exposure</string>
                   </property>
                   <property name="buddy">
                    <cstring>line_exposure</cstring>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QLineEdit" name="line_exposure"/>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="label_5">
                   <property name="text">
                    <string>Gain</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QLineEdit" name="line_gain"/>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="label_6">
                   <property name="text">
                    <string>Frame Rate</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QLineEdit" name="line_framerate"/>
                 </item>
                 <item row="3" column="0">
                  <widget class="QLabel" name="label_7">
                   <property name="text">
                    <string>Width</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="1">
                  <widget class="QLineEdit" name="line_width"/>
                 </item>
                 <item row="4" column="0">
                  <widget class="QLabel" name="label_8">
                   <property name="text">
                    <string>Height</string>
                   </property>
                  </widget>
                 </item>
                 <item row="4" column="1">
                  <widget class="QLineEdit" name="line_height"/>
                 </item>
                 <item row="5" column="0">
                  <widget class="QLabel" name="label_9">
                   <property name="text">
                    <string>X-Center</string>
                   </property>
                  </widget>
                 </item>
                 <item row="5" column="1">
                  <widget class="QLineEdit" name="line_xcenter"/>
                 </item>
                 <item row="6" column="0">
                  <widget class="QLabel" name="label_10">
                   <property name="text">
                    <string>Y-Center</string>
                   </property>
                  </widget>
                 </item>
                 <item row="6" column="1">
                  <widget class="QLineEdit" name="line_ycenter"/>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBox_2">
                <property name="title">
                 <string>DAQ</string>
                </property>
                <layout class="QFormLayout" name="formLayout_2">
                 <item row="0" column="0">
                  <widget class="QLabel" name="label_11">
                   <property name="text">
                    <string>Frequency W1</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QLineEdit" name="line_frequencyw1"/>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="label_12">
                   <property name="text">
                    <string>Amplitude W1</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QLineEdit" name="line_amplitudew1"/>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="label_13">
                   <property name="text">
                    <string>Frequency W2</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QLineEdit" name="line_frequencyw2"/>
                 </item>
                 <item row="3" column="0">
                  <widget class="QLabel" name="label_14">
                   <property name="text">
                    <string>Amplitude W2</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="1">
                  <widget class="QLineEdit" name="line_amplitudew2"/>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBox_3">
                <property name="title">
                 <string>Data</string>
                </property>
                <layout class="QFormLayout" name="formLayout_4">
                 <item row="1" column="1">
                  <widget class="QLineEdit" name="line_filename"/>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="label_15">
                   <property name="text">
                    <string>Filename</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="label_16">
                   <property name="text">
                    <string>Total Frames</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QLineEdit" name="line_totalframes"/>
                 </item>
                 <item row="0" column="1">
                  <widget class="QLineEdit" name="line_mincycles"/>
                 </item>
                 <item row="0" column="0">
                  <widget class="QLabel" name="label_17">
                   <property name="text">
                    <string>Min Cycles</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="0">
                  <widget class="QPushButton" name="button_start">
                   <property name="text">
                    <string>Start</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="1">
                  <widget class="QPushButton" name="button_stop">
                   <property name="text">
                    <string>Stop</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QWidget" name="widget" native="true">
             <property name="minimumSize">
              <size>
               <width>400</width>
               <height>400</height>
              </size>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="ImageView" name="widget_6" native="true"/>
              </item>
              <item>
               <widget class="ImageView" name="widget_2" native="true"/>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1512</width>
     <height>37</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_open"/>
    <addaction name="actionExport_Data"/>
   </widget>
   <widget class="QMenu" name="menuCa_era">
    <property name="title">
     <string>Camera</string>
    </property>
    <addaction name="action_connect"/>
    <addaction name="action_close"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuCa_era"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_open"/>
  </widget>
  <action name="action_open">
   <property name="icon">
    <iconset>
     <normaloff>../../../../../../Downloads/Win11-red/Win11-red/actions/16/document-open.svg</normaloff>../../../../../../Downloads/Win11-red/Win11-red/actions/16/document-open.svg</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExport_Data">
   <property name="text">
    <string>Export Data</string>
   </property>
  </action>
  <action name="action_connect">
   <property name="text">
    <string>Connect</string>
   </property>
  </action>
  <action name="action_close">
   <property name="text">
    <string>Disconnect</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ImageView</class>
   <extends>QWidget</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
